### PR TITLE
Fixes wrong selected domain name

### PR DESCRIPTION
**Pull request by belcherj**
#### Changes proposed in this Pull Request

* Fixes: p2MSmN-6Vk-p2

#### Testing instructions

* Have site with custom domain and no G Suite
* Have *.wordpress.com as primary domain
* Goto domains and click on custom domain
* Click "email"
* Click to add G Suite
* Add user
* Goto checkout and observe that domain is correct in the left column

### CODE REVIEW

**Comment by stephanethomas** [comment]
I'll address my comments, and merge it.

**Comment by stephanethomas** [comment]
I'm afraid we'll have to iterate a bit more on that one, I've just noticed that the domain name displayed on the `Add G Suite` page is not the right one:

screenshot

I bet we trigger a Tracks event for the `Learn more about integrating G Suite with your site.` link with the wrong domain name as well.

File: client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx [hunk 1]
```diff
@@ -98,7 +98,10 @@ class AddEmailAddressesCard extends React.Component {
 	getNewFieldset() {
 		let domain;
 
-		if ( this.props.selectedDomainName ) {
+		if (
+			this.props.selectedDomainName &&
+			getGoogleAppsSupportedDomains( [ this.props.selectedDomainName ] ).length
```
[the remark this answers is not in the document]
Comment (belcherj) [reply client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx]: I don't think that helps us.  Domains needs to be loaded for that.  Additionally we just want to check if the domain is valid for G Suite.  I think we can make better in a refactor.

[the remark this answers is not in the document]
Comment (stephanethomas) [reply client/my-sites/domains/domain-management/add-google-apps/add-email-addresses-card.jsx]: I'm not sure to understand what you mean. I was just suggesting the following change:

```javascript
if (
    this.props.selectedDomainName &&
    hasGoogleAppsSupportedDomain( [ this.props.selectedDomainName ] )
) {
```

**Comment by stephanethomas** [comment]
> I've just noticed that the domain name displayed on the Add G Suite page is not the right one:

This is still happening when the primary address of the site is set to the free `.wordpress.com` address. Sorry, I should have mentioned that.

**Comment by stephanethomas** [comment]
I've added a commit to retrieve and display the first domain that is eligible to G Suite.

**Comment by belcherj** [comment]
screen shot 2019-01-02 at 10 01 47 am

Unfortunately this will pull in the latest site and not use the selected site if it exists.

**Comment by robertf4** [comment]
LTGM

**Pull request merged**